apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -366,47 +396,73 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
       // Bump step and reset substep.
       currentProgress.step++;
       currentProgress.subStep = 0;
-    } else {
-      // Leave these values intact, and go to the results.
-      setMode(Mode.RESULTS);
     }
 
     return currentProgress;
   };
 
-  // Handle moments during generation when we should play a sound.
+  // Handle moments when we should play a sound or do another action.
   useEffect(() => {
-    if (mode === Mode.GENERATING && generatingProgress.subStep === 2) {
+    if (
+      mode === Mode.GENERATING &&
+      generatingProgress.subStep === GENERATING_SUBSTEP_COUNT - 1
+    ) {
       if (generatingProgress.step < BAD_GENERATED_RESULTS_COUNT) {
-        playSound('ai-generate-no', {volume: 0.25});
+        playSound('ai-generate-no', {volume: 0.2});
       } else {
-        playSound('ai-generate-yes', {volume: 0.25});
+        setMode(Mode.GENERATED);
+      }
+    } else if (mode === Mode.GENERATED) {
+      if (generatedProgress === 1) {
+        playSound('ai-generate-yes', {volume: 0.2});
+      } else if (generatedProgress === GENERATED_STEPS_COUNT - 1) {
+        setMode(Mode.RESULTS);
       }
     }
-  }, [generatingProgress, mode, playSound]);
+  }, [generatingProgress, generatedProgress, mode, playSound]);
+
+  // Linear interpolation.
+  const lerp = (step: number, steps: number, min: number, max: number) => {
+    return (step / steps) * (max - min) + min;
+  };
```

Comment:
nit: mind moving this up near other util functions (like `getRangeArray`) for easier extraction later?

Reply:
Good call, updated.